### PR TITLE
Improved Ecosystem app handling

### PR DESCRIPTION
**Pull request by partyparrotgreg**
## What is the purpose of the change?

The purpose is to centralise the way the ecosystem apps are managed and shown on the documentation pages.

## Describe the changes to the documentation

- New endpoint for categorised ecosystem apps that can be used in different sections of the docs
- New components to display AppCard and grids

### CODE REVIEW

File: components/AppCard/AppCardsGridCategory.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from 'react';
+import { EcosystemDocsCategory, EcosystemResponse, getSeiEcosystemAppByCategory } from '../../data/ecosystemData';
+import { EcosystemSkeleton } from '../EcosystemMap';
+import AppCardV2 from './AppCard.v2';
+
+// TODO: need to operate on the description field and then display small grid of apps
```
Comment (besated) [inline components/AppCard/AppCardsGridCategory.tsx]: is this todo still relevant?

File: data/ecosystemData.ts [hunk 2]
```diff
@@ -31,7 +35,7 @@ export type EcosystemResponse = {
 };
 
 export async function getSeiEcosystemAppsData(): Promise<EcosystemResponse> {
-	const url = 'https://app-api.seinetwork.io/webflow/ecosystem';
+	const url = 'http://app-api.seinetwork.io/webflow/ecosystem'; // TODO: Move to ENV
```
Comment (besated) [inline data/ecosystemData.ts]: why is this not `https`?

File: data/ecosystemData.ts [hunk 3]
```diff
@@ -45,10 +49,34 @@ export async function getSeiEcosystemAppsData(): Promise<EcosystemResponse> {
...
+	try {
+		const response = await fetch(url, {
+			method: 'GET',
+			headers
+		});
+
+		if (!response.ok) {
+			throw new Error(`HTTP error! Status: ${response.status}`);
+		}
+		if (!response.ok) {
```
Comment (besated) [inline data/ecosystemData.ts]: there's duplicate checks for `response.ok`

File: next.config.js [hunk 4]
```diff
@@ -10,6 +10,16 @@ module.exports = withNextra({
...
 				pathname: '**'
+			},
+			{
+				protocol: 'https',
+				hostname: 'uploads-ssl.webflow.com',
+				pathname: '**'
+			},
+			{
+				protocol: 'https',
+				hostname: '*',
```
Comment (besated) [inline next.config.js]: why is there a wildcard here for all hostnames? this could open the app up to potential vulnerabilities

Comment (partyparrotgreg) [reply next.config.js]: that was to handle logo images but I see they are all hosted on cdn.prod.website-files.com so going to update the hostname

File: next.config.js [hunk 5]
```diff
@@ -10,6 +10,16 @@ module.exports = withNextra({
...
+			},
+			{
+				protocol: 'https',
+				hostname: 'uploads-ssl.webflow.com',
+				pathname: '**'
+			},
+			{
+				protocol: 'https',
+				hostname: 'cdn.prod.website-files.com',
+				pathname: '**'
```
Comment (besated) [inline next.config.js]: can we add a more specific `pathname` to these domains here? they both currently allow all webflow cdn assets to be loaded, which is too broad

looking at the cms responses, it seems like we can do: `pathname: '/65cb43fecf24523357feada9/**',`. Example url: https://cdn.prod.website-files.com/65cb43fecf24523357feada9/66daf1b78e673eff3f8c790c_66c64307f6eb368073ba89b2_UNhif5Wy_400x400.jpeg

**Review by besated — APPROVED** [review]

**Pull request merged**